src/NuGet.Services.AzureSearch/SearchService/SearchParametersBuilder.cs

```diff
@@ -37,7 +37,7 @@ public SearchParameters LastCommitTimestamp()
             };
         }
 
-        public SearchParameters V2Search(V2SearchRequest request)
+        public SearchParameters V2Search(V2SearchRequest request, string text)
```

Comment:
I think it would be clearer to make the parameter here `isDefaultSearch` or `excludePackagesHiddenByDefault` and move the logic up to `IndexOperationBuilder`. The idea with `IndexOperationBuilder` is that he knows the big picture and he calls into `SearchParametersBuilder` with most decisions already made. Said another way, `SearchParametersBuilder` shouldn't know anything about the search text.